avahi-core/server.c

```diff
@@ -225,26 +227,33 @@ static void incoming_probe(AvahiServer *s, AvahiRecord *record, AvahiInterface *
     }
 
     if (!ours) {
+	char address[AVAHI_ADDRESS_STR_MAX] = "(unknown)";
+        const char *iface = i->hardware ? i->hardware->name : "(unknown)";
         char *t = avahi_record_to_string(record);
 
+	avahi_address_snprint(address, sizeof(address), a);
+
         if (won)
-            avahi_log_debug("Received conflicting probe [%s]. Local host won.", t);
+            avahi_log_debug("Received conflicting probe [%s] from %s(%s). Local host won.",
+                            t, address, iface);
         else if (lost) {
-            avahi_log_debug("Received conflicting probe [%s]. Local host lost. Withdrawing.", t);
+            avahi_log_info("Received conflicting probe [%s] from %s(%s). Local host lost. Withdrawing.",
+                           t, address, iface);
```

Comment:
Without this patch avahi logs something like
```
Received conflicting record [C.local    IN      AAAA fe80::3430:b6ff:fe30:94d7 ; ttl=120] with local record to be. Withdrawing.
```
where the address can be seen so the "from" part seems to be redundant. I'm not sure about switching to "info" either. It would just flood the logs until that bug is fixed.

Reply:
That is something completely different address. From address is address of remote computer, which owns the name I am trying to register. Whereas conflicting record contains only my own address with name I am trying to register. That is not useful for debugging. I need information what machine won the registration of the proposed name.

Existing non-debug log shows just my new name, but no information with who I have clashed. I think such information is vital for analysis, why my machine conflicts with itself.

What I have used for testing is:
- Have other machine having avahi with a name, say ``test-a.local``. 
- My machine has name ``test-b.local``.
- call ``sudo avahi-set-host-name test-a`` on my machine and observe, what is logged.
- addresses in record and from should be different in this case.

Reply:
> That is completely different address

I'm looking at
```
Received conflicting record [C.local    IN      AAAA fe80::3430:b6ff:fe30:94d7 ; ttl=120] with local record to be from fe80::3430:b6ff:fe30:94d7(ve-TEST-01). 
Withdrawing.
```
and the from address is the same here and it shows the address of the remote machine.

Reply:
What I get is:

```
Received conflicting record [menryzen.local	IN	AAAA 2a03:3b40:296:0:9b07:171c:858a:ffbb ; ttl=60] with local record to be from fe80::be0d:7fb0:1317:9d59(enp0s31f6). Withdrawing.

$ avahi-resolve -a fe80::be0d:7fb0:1317:9d59 
fe80::be0d:7fb0:1317:9d59	menryzen.local
$ avahi-resolve -a 2a03:3b40:296:0:9b07:171c:858a:ffbb
2a03:3b40:296:0:9b07:171c:858a:ffbb	menryzen.local
```

Ah, I did not expect both of those addresses are of the remote machine. They are just different in my case. Okay, you are correct that the another address is not useful. Interface information could be. Since it prints into info severity lines like ``Withdrawing address record for ::1 on lo.`` and ``Registering new address record for ::1 on lo.*.`` after name change, I think more information about conflict should be logged in non-debug environment as well.

Reply:
From address should be the actual source address of a packet, from which packet were received on socket level. I think the record printed contains addresses from that host, which might conflict. Not sure, they could be potentially different. Especially in some reflection scenarios.

Reply:
> Registering new address record for ::1 on lo.*.

I think in the context of those messages interface names are helpful because they show what avahi registers and where. In the context of conflicting records I'm not sure interfaces matter that much probably.

> Since it prints into info severity lines like

If it logged that with the debug info there would be no way to figure out what's happening by just looking at the logs so the info level is more or less justified. With name conflicts it's different.

Generally I'm on the fence about this. If I get to the point where I have to debug name conflicts I can probably turn on the debug level. It should be possible to run Wireshark too. I can see where messages like
```
Host name conflict with some machine, retrying with C
```
can be helpful but I think the "conflicting record" messages are detailed enough probably.

Reply:
I am not so sure. I have not seen any checks on the incoming message, that at least on of addresses in incoming DNS packet belongs to the sender of that packet. We may conflict with some proxy, but we need to know address of that proxy in this case. Not addresses inside the message, which might be actually ours.

Unfortunately not enough details are passed to ``Host name conflicts`` message in *server_callback* in avahi/daemon/main.c. It would take a lot of parameters adding. So I have taken a bit shortcut by logging those information earlier, when we still have the information.

By the way, logging of conflicts are quite inconsistent:

```
$ grep 'Host name conflict' -r avahi*
avahi-core/avahi-test.c:        avahi_log_debug("Host name conflict, retrying with <%s>", n);
avahi-core/conformance-test.c:        avahi_log_warn("Host name conflict, retrying with <%s>", n);
avahi-daemon/static-hosts.c:            avahi_log_error("Host name conflict for \"%s\", not established.", h->host);
avahi-daemon/main.c:            avahi_log_warn("Host name conflict, retrying with %s", n);
avahi-daemon/main.c:            sd_notifyf(0, "STATUS=Host name conflict, retrying with %s", n);
avahi-utils/avahi-publish.c:                fprintf(stderr, _("Host name conflict\n"));
```

Notice always a different importance is used. Perhaps we should unify it in different tools?

Reply:
> I have not seen any checks on the incoming message, that at least one of addresses in incoming DNS packet belongs to the sender of that packet. We may conflict with some mdns proxy, but we need to know address of that proxy in this case

I still think that if I get to this point I have to use debuggers, run Wrieshark and so on. I can't see how logs can help with anything.

> Notice always a different importance is used. Perhaps we should unify it in different tools?

Those messages are logged in different contexts. The tests don't matter generally. In `static-hosts` it's "error" because avahi doesn't keep retrying with new names as far as I can remember. In `main.c` it's "warning" because conflicts don't prevent avahi from getting another name. `avahi-publish` uses the "verbose" switch so log levels don't matter there and it just crashes when conflicts occur https://github.com/avahi/avahi/issues/478. It would be nice to make logs consistent in general of course but it all depends on the context so switching everything to "info" isn't ideal either.